Fix compiling with shared liblzf library by not mangling the needed symbols
--- a/src/VBox/Runtime/common/zip/zip.cpp	2022-10-07 01:11:55.000000000 +0800
+++ b/src/VBox/Runtime/common/zip/zip.cpp	2022-10-19 22:35:37.996450458 +0800
@@ -63,7 +63,9 @@
 #endif
 #ifdef RTZIP_USE_LZF
  RT_C_DECLS_BEGIN
+extern "C" {
 #  include <lzf.h>
+}
  RT_C_DECLS_END
 # include <iprt/crc.h>
 #endif
